### PR TITLE
Refactor deprecations in MauticCrm bundle (M3 migration)

### PR DESCRIPTION
**Pull request by anton-vlasenko**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8011
| BC breaks? | 1. Deprecated method MauticPlugin\MauticCrmBundle\Integration\SalesforceIntegration::amendToSfFields removed. 2. Deprecated method MauticPlugin\MauticCrmBundle\Integration\SugarcrmIntegration::amendToSfFields removed
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR containst changes that are neccessary to make MauticCrmBundle compatible with Symfony3.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to Plugins, try to authenticate ConnectWise plugin.
3. Optionally go to Plugin and try to open (probably not worth to authorize, it's hard to get all the credentials) the following integrations: Dynamics CRM, Hubspot, Pipedrive, Salesforce, Sugar CRM, Vtiger, Zoho.
3. If you are able to authorize ConnectWise and you are able to open popups with the integrations mentioned above then we can count this task as done.

### CODE REVIEW

File: plugins/MauticCrmBundle/Form/Type/IntegrationCampaignsTaskType.php [hunk 1]
```diff
@@ -89,15 +89,15 @@ function ($validateMe, ExecutionContextInterface $context) {
     /**
      * {@inheritdoc}
      */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
```
Comment (escopecz) [inline plugins/MauticCrmBundle/Form/Type/IntegrationCampaignsTaskType.php]: Use `array_flip()` on the member choices on line 71 and ensure that all choice type choices in this bundle have the values and labels flipped.

Comment (anton-vlasenko) [reply plugins/MauticCrmBundle/Form/Type/IntegrationCampaignsTaskType.php]: Fixed in eb4f00a1e0a52ca435769071b7a18dda1d225b1f

Comment (escopecz) [reply plugins/MauticCrmBundle/Form/Type/IntegrationCampaignsTaskType.php]: I forgot to tell you that you should also add `choices_as_values` option as described in https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

I thought I added it to all comments on this topic I did yesterday but I must have forgotten it here.

Comment (anton-vlasenko) [reply plugins/MauticCrmBundle/Form/Type/IntegrationCampaignsTaskType.php]: No worries. Fixed in bfc595560a387c4015f4f14fd3992c57679f83cf

**Comment by escopecz** [comment]
I loaded config forms of all CRM integrations and they loaded correctly 👍

**Pull request merged**